packages/validation/src/permissions/research-output.ts

```diff
@@ -1,92 +1,50 @@
 import { User } from '@asap-hub/auth';
-import {
-  ResearchOutputPostRequest,
-  ResearchOutputResponse,
-  ResearchOutputWorkingGroupResponse,
-  userPermissions,
-  WorkingGroupDataObject,
-} from '@asap-hub/model';
-import { isResearchOutputWorkingGroup } from '../research-output-guards';
+import { UserPermissions, UserResponse } from '@asap-hub/model';
 
-export const hasCreateUpdateResearchOutputPermissions = (
-  user: Omit<User, 'algoliaApiKey'>,
-  teams: ResearchOutputPostRequest['teams'],
-): boolean =>
-  !!user.teams.find((team) => {
-    if (team.role === 'ASAP Staff') {
-      return true;
-    }
-
-    return teams.includes(team.id) && team.role === 'Project Manager';
-  });
-
-export const hasWorkingGroupsCreateUpdateResearchOutputPermissions = (
-  user: Omit<User, 'algoliaApiKey'>,
-  workingGroup: WorkingGroupDataObject | undefined,
-): boolean => {
-  if (!workingGroup) return false;
-
-  const { leaders } = workingGroup;
+export const noPermissions: UserPermissions = {
+  saveDraft: false,
+  publish: false,
+};
 
-  return leaders.some((leader) => {
-    if (leader.user.id === user.id && leader.role === 'Project Manager') {
-      return true;
-    }
-    return false;
-  });
+export const partialPermissions: UserPermissions = {
+  saveDraft: true,
+  publish: false,
 };
 
-export const noPermissions: userPermissions = {
-  createDraft: false,
-  editDraft: false,
-  publishDraft: false,
-  editPublished: false,
+export const fullPermissions: UserPermissions = {
+  saveDraft: true,
+  publish: true,
 };
 
-export const isUserProjectManagerOfTeams = (
-  user: Omit<User, 'algoliaApiKey'>,
-  teams: ResearchOutputResponse['teams'],
-): boolean =>
-  !!user.teams.find(
-    (team) =>
-      teams.map((t) => t.id).includes(team.id) &&
-      team.role === 'Project Manager',
-  );
+export const getUserPermissions = (
+  user: Omit<User, 'algoliaApiKey'> | UserResponse | null,
+  entity: 'teams' | 'workingGroups',
+  entityIds: string[],
+): UserPermissions => {
```

Comment:
I see what you did here - you're trying to make this permission check more generic, but you're shifting the responsibility of saying what is allowed to the client of this function now. Imho we should keep those functions as they were, returning booleans, and if you need to check whether the user has permissions to save a draft - add another function. This should be composable anyway so it's not like it's going to be a huge repetition?